src/scripts/datadog-install.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+export DEBIAN_FRONTEND=noninteractive
+
+ELASTIC_CONFIG_FILE=/etc/datadog-agent/conf.d/elastic.d/elastic.yaml
+DD_AGENT=/var/tmp/install_datadog_agent.sh
+
+# Custom logging with time so we can easily relate running times, also log to separate file so order is guaranteed.
+# The Script extension output the stdout/err buffer in intervals with duplicates.
+log()
+{
+    echo \[$(date +%d%m%Y-%H:%M:%S)\] "$1"
+    echo \[$(date +%d%m%Y-%H:%M:%S)\] "$1" >> /var/log/datadog-install.log
+}
+
+if [ "$1" == "" ]; then
+    log "Must provide a DataDog API Key as the one and only argument" >&2
+    exit 1
+fi
+
+log "Installing DataDog plugin onto this machine using API Key [$1]"
+
+wget -q https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh -O $DD_AGENT
+chmod +x $DD_AGENT
+DD_API_KEY=$1 $DD_AGENT
+rm /etc/datadog-agent/conf.d/elastic.d/auto_conf.yaml
+
+echo "ad_identifiers:
+    - elasticsearch
+instances:
+    - url: 'http://localhost:9200'
+pshard_stats: true
+cluster_stats: false
+pending_task_stats: true
+tags:
+    - 'elasticsearch-role:data-node'" >> $ELASTIC_CONFIG_FILE
```

Comment:
Just noticed. We need to adjust the role - data / client / master

Reply:
Oh crap, so do we need to go back to not using the `commonInstallParams` so we can pass in this role value in each instance of the setup?

Reply:
I think we can still get away with this - if you look at the main script, it takes either `-x`,  `-y`, or `-z` to set the node role, and since we call this script from the main one, we can use that info.

Reply:
@tyrotoxin hey serge, I think I fixed this in a decent way. The current way used simple bit flags for MASTER_NODE_ONLY, DATA_NODE_ONLY, CLIENT_NODE_ONLY which means it would have been awkward to pass those on through to the underlying datadog script. So I added a new variable called NODE_TYPE that sets it to either master, client, or data and then passes that down into the datadog script for use in the config file.